### PR TITLE
always replace existing installations with `kernelspec.install`

### PR DESCRIPTION
**Pull request by minrk**
deprecate replace arg with a DeprecationWarning that nobody will ever see...

Also return the install destination for use by callers.

### CODE REVIEW

**Comment by takluyver** [comment]
Seems fine

**Comment by takluyver** [comment]
There's a test that OSError is raised when you try to install a kernelspec that already exists.

File: jupyter_client/kernelspec.py [hunk 1]
```diff
@@ -143,22 +143,27 @@ def install_kernel_spec(self, source_dir, kernel_name=None, user=False,
...
         if not kernel_name:
             kernel_name = os.path.basename(source_dir)
         kernel_name = kernel_name.lower()
-
+        
+        if replace is not None:
+            warnings.warn(
+                "replace is ignored. Installing a kernelspec always replaces an existing installation",
+                DeprecationWarning,
+            )
```
Comment (Carreau) [inline jupyter_client/kernelspec.py]: stacklevel=2 to actually have the error message pointing at the function call ?

**Comment by Carreau** [comment]
Test need fixing. 

+1

**Comment by minrk** [comment]
test should be fixed, warning stack level changed

**Pull request merged**